takakonishimura/project-reclass
# Terraform & Packer

**Goal:**
Automate the deployment of EC2 instances and creation of their base AMIs such that onboarding new members to a working development environment. See new `project-reclass/infra/toynet/README.md` for workflow details when adding a user or updating AMI images.

**Testing:**
[Setup AWS credentials](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/getting-your-credentials.html), [store the credential](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html) locally, and `export AWS_PROFILE=default`.

1) Go to `project-reclass/infra/toynet/packer` and run `packer build toynet.json`
2) Take note of the new ami_id
3) Go to `project-reclass/infra/toynet/terraform` and run `terraform init`
4) Alter 'newUser.tfvars` to contain `ami_id = <your new ami id from packer>`
5) Run `terraform plan`
6) See that 2 instances will be destroyed and 2 more created
7) Run `terraform apply`
8) Go to the AWS console and confirm that the new instances can have the `project-reclass`
9) Go to `project-reclass/toynet` and execute `./run_toynet.sh`
10) See a successful toynet output image.

**Longterm Goals:**
* Move to docker so I don't need to host dev environments
* Read up on permission model in AWS
